Assets/BossRoom/Scripts/Shared/Net/ConnectionManagement/GameNetPortal.cs

```diff
@@ -245,8 +247,27 @@ public void RequestDisconnect()
             }
             m_ClientPortal.OnUserDisconnectRequest();
             m_ServerPortal.OnUserDisconnectRequest();
+
+            if (NetManager.IsServer)
+            {
+                StartCoroutine(WaitToShutdown());
+            }
+            else
+            {
+                ShutDown();
+            }
+        }
+
+        void ShutDown()
+        {
             SessionManager<SessionPlayerData>.Instance.OnUserDisconnectRequest();
             NetManager.Shutdown();
         }
+
+        IEnumerator WaitToShutdown()
+        {
+            yield return new WaitForSeconds(0.5f);
```

Comment:
ouch, why?

Reply:
Same reason as when we do it before disconnecting a client when approval check fails. If we don't, Netcode will disconnect the clients before we can send them a message telling them why.

Reply:
thing is, we don't send here, we send elsewhere. The wait should be before the send of the custom message, not here then.

Reply:
I added it here since this is where the Shutdown was, but I could move it to the ClientGameNetPortal and ServerGameNetPortal if it makes it clearer

Reply:
if you plan to reuse any of this code for develop, then yeah, let's make this clearer.